src/components/Documentation/Markdown/index.tsx

```diff
@@ -30,17 +36,31 @@ const isInsideCodeBlock = (node: Element): boolean => {
 }
 
 const Details: React.FC<{
-  children: Array<{ props: { children: Array<string> } } | string>
+  children: Array<{ props: { children: ReactNode } } | string>
 }> = ({ children }) => {
   const filteredChildren = children.filter(child => child !== '\n')
 
   if (!filteredChildren.length) return null
   if (typeof filteredChildren[0] === 'string') return null
 
-  const text = filteredChildren[0].props.children[0]
+  const headingChildren: ReactNode[] = filteredChildren[0].props
+    .children as ReactNode[]
+
+  // Remove header auto-link if present
+  const finalHeadingChild = headingChildren[headingChildren.length - 1] as {
+    props: {
+      className: string
+    }
+  }
+
+  const triggerChildren: unknown =
+    finalHeadingChild.props !== undefined &&
```

Comment:
how can it be undefined btw?

Reply:
Plaintext children in React children are represented by strings which don't have a `props` like component children.